### PR TITLE
NMAP Dump Payload & Recon iSpy Payload

### PR DESCRIPTION
**Pull request by infoskirmish**
See readme.md and comments in payload.sh.

### CODE REVIEW

**Comment by sundhaug92** [comment]
Nice. Two questions? Why NAT and not TRANSPARENT and what's the difference between this and nmapper (#8)?

**Comment by infoskirmish** [comment]
Great question. 
1) I have submitted an update as my orig. pull request got an old file... still new to git. So please check out the latest.

2) Nmap Dump is different in that it has some basic logic built in. It will scan the various interfaces, find ones that it can send data out on.

3) Why NAT? I have added a global variable where the user can change it to Transparent or Bridge or whatever suits their needs.

4) The payload will find ip addresses based on interfaces that are up (regardless of the network mode). The script will automatically convert these ip address into subnets that then nmap will scan.

5) I log useful details as the payload is executing that I find very helpful in trouble shooting issues.  The log is in /mnt/loot/nmapdump/log.txt

6) I have the code well commented

7) The user can choose what the payload will do after executing: halt, reboot, nothing, 

8) Darren has mentioned before he would like to see payloads with some logic checks built in; this payload does that. 

9) It's awesome.

**Comment by thehappydinoa** [comment]
@sundhaug92 @infoskirmish I honestly think this payload is better than the one I created, I will close my PR

**Comment by infoskirmish** [comment]
Thank you very much for the kind words.

**Comment by infoskirmish** [comment]
The "iSpy Intel Dump" was supposed to a separate pull request as it is a completely different payload. My git noob self despite putting it in a different payload folder can't figure out how to not get it to sync with this pull request. #headdesk

**Comment by sundhaug92** [comment]
Fair @thehappydinoa, just thought it'd be an interesting comparison-point

**Comment by sundhaug92** [comment]
@infoskirmish It's OK, you could change the PR-name

File: payloads/library/sniffing/nmapdump/payload.sh [hunk 1]
```diff
@@ -0,0 +1,263 @@
+#!/bin/bash
+# 
+# Title:		NMap Dump
+# Description:		Dumps NMap scan data to USB storage.
+# Author: 		infoskirmish.com
+# Version:		1.0
+# Category:		sniffing
+# Target: 		Any
+# Net Mode:		NAT
```
Comment (sundhaug92) [inline payloads/library/sniffing/nmapdump/payload.sh]: Doesn't match line 26

**Comment by sundhaug92** [comment]
What testing have you done on the payloads?

**Comment by infoskirmish** [comment]
@sundhaug92 I have tested the payload in simulated deployments on my lab LAN. With various net modes and using preset interfaces to letting the payload cycle through all the interfaces.

**Comment by sundhaug92** [comment]
Ok. Looks good, only thing is the unnecessary HTML-tags and the usage of HTML instead of markdown but that's nitpicking

**Comment by thehappydinoa** [comment]
@infoskirmish Next time you make a PR try to make different branches for different PR's that way you can track progress separately. Also, this payload is scary amazing.

**Comment by cevans3505** [comment]
I'm not sure if I'm more impressed by the payload or the beauty of well formatted and commented code. Kudos. I will play around with these later this evening.

**Comment by infoskirmish** [comment]
@thehappydinoa Won't that add another repo to my lists of repos? Thank you for the feedback!

**Comment by infoskirmish** [comment]
@cevans3505 I try to devote time to making code look good. It is almost like art. Enjoy and please give me any feedback you have.

**Comment by infoskirmish** [comment]
@sundhaug92 are you talking about the html in the README.md of the iSpy module? If so, just out of curiosity why is markdown better than html?

**Comment by thehappydinoa** [comment]
@infoskirmish It should not, a branch is within one repo. Often developers use branches for development before pushing un-tested software. Community use them too its like using kali-dev rather than kali-rolling they are branches of the same repo.

**Comment by infoskirmish** [comment]
@sundhaug92 updated README.md to remove html and use markdown. :)

**Comment by sundhaug92** [comment]
@infoskirmish Well, markdown is a kind of a superset of HTML intended to reduce boilerplate. In this case it's not because markdown is better in anyway but because it's what everyone else does in these payloads-repos. It's nice to see you use it though.

**Comment by sundhaug92** [comment]
Why close?

**Comment by infoskirmish** [comment]
I separated them into different pull requests. See the open pull requests.

**Pull request closed without merging**